app/(interview)/onboard/[protocolId]/route.ts

```diff
@@ -11,7 +12,7 @@ const handler = async (
   { params }: { params: { protocolId: string } },
 ) => {
   const protocolId = params.protocolId; // From route segment
-  const url = req.nextUrl.clone();
+  const url = env.PUBLIC_URL ? new URL(env.PUBLIC_URL) : req.nextUrl.clone();
```

Comment:
Needs a comment to explain it, and perhaps better written with optional chaining rather then ternary.